[receiver/filelog] Flaky test - TestStorage

```
=== CONT  TestStorage
    testing.go:967: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestStorage1460386472\001\test.log: The process cannot access the file because it is being used by another process.
--- FAIL: TestStorage (0.12s)
```

Observed here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/6760041345?check_suite_focus=true#step:7:1588

--- a/receiver/filelogreceiver/storage_test.go
+++ b/receiver/filelogreceiver/storage_test.go
@@ -144,9 +144,11 @@ func TestStorage(t *testing.T) {
 	for _, e := range host.GetExtensions() {
 		require.NoError(t, e.Shutdown(ctx))
 	}
+	require.NoError(t, logger.close())
 }
 
 type recallLogger struct {
+	logFile *os.File
 	*log.Logger
 	written []string
 }
@@ -157,6 +159,7 @@ func newRecallLogger(t *testing.T, tempDir string) *recallLogger {
 	require.NoError(t, err)
 
 	return &recallLogger{
+		logFile: logFile,
 		Logger:  log.New(logFile, "", 0),
 		written: []string{},
 	}
@@ -170,6 +173,10 @@ func (l *recallLogger) log(s string) {
 func (l *recallLogger) recall() []string {
 	defer func() { l.written = []string{} }()
 	return l.written
+}
+
+func (l *recallLogger) close() error {
+	return l.logFile.Close()
 }
 
 // TODO use stateless Convert() from #3125 to generate exact plog.Logs

